flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/BufferConsumer.java

```diff
@@ -118,6 +118,16 @@ public BufferConsumer copy() {
 		return new BufferConsumer(buffer.retainBuffer(), writerPosition.positionMarker, currentReaderPosition);
 	}
 
+	/**
+	 * Returns a retained copy with separate indexes and syncs it with the current position of the given {@link BufferBuilder}.
+	 * This allows to read from the same {@link MemorySegment} twice starting from the position of the builder.
+	 *
+	 * @return a retained copy of self with separate indexes
+	 */
+	public BufferConsumer copyAndSync(BufferBuilder builder) {
```

Comment:
How about renaming it as `copyWithReaderPosition(int readerPosition)`? The semantic of this method is for copying the current consumer with setting any reader position which is not only limited to `builder.getCommittedBytes()`, and then we can pass the `builder.getCommittedBytes()` from the caller by different usages.

Another consideration is that only build creates the respective consumer in single direction, so consumer will not see the builder before. `copyWithReaderPosition(int readerPosition)` can avoid touching builder somehow from public way.

Reply:
I like it as it gives more flexibility, I just need to update the writer position publicly, but imho that is a good change anyway (needed that we debugging in various places).